openstf/model/train.py

```diff
@@ -160,30 +243,93 @@ def train_model_for_specific_pj(
         model_trainer.confidence_interval,
     )
     train_predict = predictor.make_forecast(
-        train_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["train_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
     validation_predict = predictor.make_forecast(
-        validation_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["validation_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
     test_predict = predictor.make_forecast(
-        test_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["test_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
 
-    context.perf_meter.checkpoint("predicting")
+    return {
+        "train_predict": train_predict,
+        "validation_predict": validation_predict,
+        "test_predict": test_predict,
+    }
+
+
+def create_evaluation_figures_for_specific_pj(
+    model_trainer, split_input_data, split_predicted_data
+):
+    """Create figures of feature importance and data series of input and predicted data.
+
+    Args:
+        model_trainer (MLModelType): model trainer
+        split_input_data (dict): dict of cleaned data with features, split in train,
+            validation and test
+        split_predicted_data (dict): dict of predicted data, split in train, validation
+            and test
+
+    Returns:
+        tuple: tuple containing:
+            plotly.graph_objects.Figure: A treemap of the features.
+            dict: dict of line plots of of split input and predicted
+                data series per prediction horizon.
+    """
 
     # Create figures
     figure_features = plot_feature_importance(model_trainer.feature_importance)
     figure_series = {
         f"Predictor{horizon}": plot_data_series(
-            [train_data, validation_data, test_data],
-            [train_predict, validation_predict, test_predict],
+            list(split_input_data.values()),
```

Comment:
Perhaps you can add the argument names here to make it more clear. If this data transformation is really required you could add a `to_list` method to your dataclass if you go that route.

Reply:
I wanted to save lines, why is that bad?

Reply:
Saving lines by itself can be a bad idea. I think readability and reducing complexity are more important then saving lines. You can write down very hard to read and complex code when you minimizing lines is your goal.

But its of course also al balancing act. In this case its a bit too much I think, too many things happen in one statement and also the list concatenating happens for every iteration in the dict comprehension.